pubsub/subscriptions.rb

```diff
@@ -0,0 +1,253 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/pubsub"
+
+def update_push_configuration project_id:, subscription_name:, new_endpoint:
+  # [START update_push_configuration]
+  # project_id        = Your Google Cloud Project ID
+  # subscription_name = Your Pubsub subscription name
+  # new_endpoint      = Endpoint where your app receives messages
+  pubsub = Google::Cloud::Pubsub.new project: project_id
+
+  subscription = pubsub.subscription subscription_name
+  subscription.endpoint= new_endpoint
+  puts "Push endpoint updated."
+  # [END update_push_configuration]
+end
+
+def list_subscriptions project_id:
+  # [START list_subscriptions]
+  # project_id = Your Google Cloud Project ID
+  pubsub = Google::Cloud::Pubsub.new project: project_id
+
+  subscriptions = pubsub.list_subscriptions
+  puts "Subscriptions:"
+  subscriptions.each do |subscription|
+    puts subscription.name
+  end
+  # [END list_subscriptions]
+end
+
+def delete_subscription project_id:, subscription_name:
+  # [START delete_subscription]
+  # project_id        = Your Google Cloud Project ID
+  # subscription_name = Your Pubsub subscription name
+  pubsub = Google::Cloud::Pubsub.new project: project_id
+
+  subscription = pubsub.subscription subscription_name
+  subscription.delete
```

Comment:
Add a new line after this line. Try to group sections of the sample based on steps.